cli/src/cli_util.rs

```diff
@@ -948,6 +949,41 @@ impl WorkspaceCommandHelper {
         }
     }
 
+    fn evaluate_revset_with_modifier_to_single_commit(
+        &self,
+        ui: &mut Ui,
+        revision_arg: &RevisionArg,
+        modifier: Option<RevsetModifier>,
+        expression: &RevsetExpressionEvaluator<'_>,
+        should_hint_about_all_prefix: bool,
+    ) -> Result<Commit, CommandError> {
+        Ok(if modifier == Some(RevsetModifier::Prompt) {
+            let targets: Vec<Commit> = expression.evaluate_to_commits()?.try_collect()?;
+            let target = match targets.as_slice() {
+                [target] => target,
+                [] => {
+                    return Err(user_error("Empty revision set"));
+                }
+                commits => choose_commit(
+                    ui,
+                    self,
+                    &format!(
+                        r#"ambiguous commit from revset "{revision_arg}", choose one to target"#
+                    ),
+                    commits,
+                )?,
+            };
+            target.to_owned()
```

Comment:
nit: it seems easier to understand if the Ok is placed inside the if-else cases.